### PR TITLE
Ogg

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: src/asset/audio.rs [hunk 1]
```diff
@@ -52,18 +75,44 @@ impl Audio {
...
         }
-        let mut samples = self.samples.write().unwrap();
         match data {
-            wav::BitDepth::ThirtyTwoFloat(data) => *samples = data,
+            wav::BitDepth::ThirtyTwoFloat(data) => data,
             _ => todo!("Only WAV containing floats are currently supported"),
         }
     }
+
+    pub fn samples_ogg(&mut self, bytes: Vec<u8>) -> Vec<f32> {
```
Comment (FredTheDino) [inline src/asset/audio.rs:100]: Isn't the thing with OGG that we can stream in the audio we need. And don't have everything sorted in ram at all times?

Is that something we want?

File: src/asset/audio.rs [hunk 2]
```diff
@@ -52,18 +75,44 @@ impl Audio {
...
+        let mut reader = lewton::inside_ogg::OggStreamReader::new(Cursor::new(&bytes)).unwrap();
+        let sample_rate = reader.ident_hdr.audio_sample_rate;
+        if sample_rate != crate::audio::SAMPLE_RATE as u32 {
```
Comment (FredTheDino) [inline src/asset/audio.rs:103]: Do we need to fix this sample rate thing?  It feels limiting.

**Review by FredTheDino — COMMENTED** [review]
LGTM - but i haven't tried it yet.

File: src/asset/audio.rs (hunk 1 above)
Comment (sornas) [reply src/asset/audio.rs:100]: Yes, that sounds nice. But, small PRs! #64

File: src/asset/audio.rs (hunk 2 above)
Comment (sornas) [reply src/asset/audio.rs:103]: #65

**Review by FredTheDino — APPROVED** [review]
LGTM!

**Pull request merged**